performance_test/helper_scripts/generate_apex_xml.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Apex.AI, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generate DEFAULT XML file for apex tests."""
+import argparse
+from xml.dom.minidom import parseString
+
+from dicttoxml import dicttoxml
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        '-w',
+        '--whitelist',
+        nargs='*',
+        help='List of whitelist interfaces separated by spaces',
+        required=True,
+    )
+    parser.add_argument(
+        '-f',
+        '--file',
+        help='Path to the file to store the XML',
+        required=True,
+    )
+    args = parser.parse_args()
+    interface_whitelist = args.whitelist
+    xml_file = args.file
+
+    profiles = {
```

Comment:
With the minidom module you can replace this dictionary with something like this:

```python
doc = Document();
transport_descriptors = doc.createElement('transport_descriptors')
transport_descriptor = doc.createElement('transport_descriptor')
transport_id = doc.createElement('transport_id')
transport_id.appendChild(doc.createTextNode('apex_transport_descriptor'))
transport_descriptor.appendChild(transport_id)
transport_descriptors.appendChild(transport_descriptor)
doc.appendChild(transport_descriptors)
```

this should be more efficient. What you're doing below is turning a dictionary into an XML document, which gets converted into a string, which is then parsed back into XML via `parseString`, and then look up the `participant` node to add an attribute, so there's a penalty every time you convert between XML and back.

Moreover, the code for adding an attribute could go right here when insert the `participant` node, instead of looking it up later.